tests/xfail/xfail.go

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package xfail
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func Enable(message string) {
+	RegisterFailHandler(func(m string, offset ...int) {
+		depth := 0
+		if len(offset) > 0 {
+			depth = offset[0]
+		}
+		Skip(fmt.Sprintf("[XFAIL] %s, failure: %s", message, m), depth+1)
+	})
+}
+
+func Disable() {
+	RegisterFailHandler(Fail)
+}
```

Comment:
We don't need Disable since we can defer to `RegisterFailHandler(Fail)` inside Enable.

And since we don't need Disable we can rename it to have something like `xfail.Reason("this and that")`

```golang
RegisterFailHandler(func(m string, offset ...int) {
                defer RegisterFailHandler(Fail)
		depth := 0
		if len(offset) > 0 {
			depth = offset[0]
		}
		Skip(fmt.Sprintf("[XFAIL] %s, failure: %s", message, m), depth+1)
	})
```

Reply:
It feels odd to change the handler from within the handler.
But this is indeed simpler for the caller.
I will try it out, thanks.

Reply:
Done.
I have left the `Enable` name as the suggested `Reason` does not seem to fit as a function name (not a verb).
If I had a better place to locate this one, under a more generic package, I would probably call it `XFail`.

Reply:
I really don't like Enable what about call package "mark" ? at pytest the annotation is something like `pytest.mark.xfail` then we can have something like 
```golang
mark.XFail("This and that")
```

Reply:
Calling the package `mark` will confuse developers IMO. Pytest can mark tests to behave differently, we do not have this generic notion here.

I thought at some point to call it `assert`, moving some helpers here like [k8s does](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/expect.go). Would you like it better?
It will then be called: `assert.XFail("reason")`.

Reply:
assert is pretty good, +1